### ISSUE TITLE
buycourses. how do I ad ideal-payment

### ISSUE THREAD

**Issue by learningcentrum** [chamilo/chamilo-lms#1840]
### Current behavior / Resultado actual / Résultat actuel
only bank and paypal
### Expected behavior / Resultado esperado / Résultat attendu
add ideal

**Comment by jloguercio** [comment]
Hi , Ideal-payment is not supported at the moment only bank transfer, paypal and culqi, but we can consider adding it in a next chamilo version.

Regards

**Comment by learningcentrum** [comment]
Thanks, How can I create a module such as Paypal and add the data is the same as that of PayPal
I made a second row in the database, this did not work, where stands it, so that I can change it, I can a little bit programeren

**Comment by ywarnier** [comment]
Hi,
The easiest process is probably to look at the difference (in the 1.11.x branch) between PayPal and Culqi in the plugin/buycourses/ directory. Culqi support was developed based on PayPal's implementation, so you should be able to extend from there to your "Ideal" payment API.

**Comment by learningcentrum** [comment]
in my version I have only PayPal and bank 
I use the: Chamilo 1.11.2 versie

greetings
Martin Doornbos

**Comment by ywarnier** [comment]
1.11.x is the development version that will result (soon) in the publication of 1.11.4.
If you want ideal to be integrated, you can either do it yourself and send us the code, or ask an official provider to do it for you as a development job: https://chamilo.org/providers

**Comment by learningcentrum** [comment]
all the information are on the site from mollie
https://www.mollie.com/nl/docs/authentication


Authentication

You hold the API keys

What data do I need?

The first thing you need is a website profile. Each website profile has a Live API key and a Test API key. Use these keys to: 
•Show the Mollie API it is really you.
•Select the website profile the intended communication with the Mollie API relates to.
•Specify whether you're testing or working with real payments.

The API key must be sent along with each API request, by providing it in the HTTP call's Authorization header. If one of our default API clients is used, the client will offer a setApiKey method that will allow you to easily configure the required header.

While building and testing your integration you should use the Test API key. This will cause your code to (automatically) only create test payments. Test payments come with a fake payment screen that allows you to select whether your test payments are successful or not without spending actual money. This way you can easily test and rehearse different scenarios that will occur later on when you switch to real payments. After your testing is done you go live by using the Live API key instead of the Test API key. Going live (or going back to testing) is as easy as changing keys. Nothing else needs to change. 

Of course it's very important to keep any API-keys secure. Do not ever share them. However, if a key leaks you can always regenerate it. Don't forget to apply new keys to your code. Until you do your integration will not work. 

Apart from the payment screen and the fact that test payments are created instead of real ones, the Mollie API behaves the same way regardless of whether the Test API key or the Live API key is used. Because of this, there won't be any technical surprises upon going live. Make a note: don't forget to start using the Live API key when your site goes public or your customers could be getting a free ride. 

The Mollie  REST API

The API implements a Representational state transfer (REST) architecture. Sounds technical but it's really quite easy. It mostly breaks down to HTTP-methods GET, POST, PUT and DELETE matching the operations to read, update, create and delete. 

REST also implies a nice and clean structure for URLs or endpoints. This means you can reach any part of the Mollie API on https://api.mollie.nl/v1/ adding the name of the resource you want to interact with. 

Example

At the top of documentation pages you can select a platform like PHP, Ruby, Python or cURL. Any code examples on the page will be shown for that platform.

Any API action you want to execute requires a valid Test API key or Live API key. In order to show you how authentication works we use the GET method on the payments Resource. This method fetches a payment, but that's not really important here. Focus on how the API-key is involved. 

In all of the code examples the Test API Key for your favorite website profile is used if you're signed in. If you're not signed in we'll use a fictional Test API Key instead. In the example below we're using the Test API Key test_dHar4XY7LxsDOtmnkVtjNVWXLSlXsM. The response shows a result for a retrieved payment with the fictional id tr_WDqYK6vllg. 



Request code 
PHP 














curl -X GET https://api.mollie.nl/v1/payments/tr_WDqYK6vllg \
    -H "Authorization: Bearer test_dHar4XY7LxsDOtmnkVtjNVWXLSlXsM"
mollie = Mollie::API::Client.new('test_dHar4XY7LxsDOtmnkVtjNVWXLSlXsM')

payment = mollie.payments.get 'tr_WDqYK6vllg'123
mollie = Mollie.API.Client()
mollie.setApiKey('test_dHar4XY7LxsDOtmnkVtjNVWXLSlXsM')

payment = mollie.payments.get('tr_WDqYK6vllg')1234
var mollie = new Mollie.API.Client;
mollie.setApiKey('test_dHar4XY7LxsDOtmnkVtjNVWXLSlXsM');

mollie.payments.get(
    'tr_WDqYK6vllg'
,   function (payment) {
    // payment
});12345678
$mollie = new Mollie_API_Client;
$mollie->setApiKey('test_dHar4XY7LxsDOtmnkVtjNVWXLSlXsM');

$payment = $mollie->payments->get('tr_WDqYK6vllg');1234


Response
HTTP/1.1 200 OK
Content-Type: application/json; charset=utf-8

{
    "resource": "payment",
    "id": "tr_WDqYK6vllg",
    "mode": "test",
    "createdDatetime": "2017-04-19T14:52:56.0Z",
    "status": "paid",
    "paidDatetime": "2014-09-05T14:37:35.0Z",
    "amount": "35.07",
    "description": "Order 33",
    "method": "ideal",
    "metadata": {
        "order_id": "33"
    },
    "details": {
        "consumerName": "Hr E G H K\u00fcppers en\/of MW M.J. K\u00fcppers-Veeneman",
        "consumerAccount": "NL53INGB0618365937",
        "consumerBic": "INGBNL2A"
    },
    "locale": "nl",
    "profileId": "pfl_QkEhN94Ba",
    "links": {
        "webhookUrl": "https://webshop.example.org/payments/webhook",
        "redirectUrl": "https://webshop.example.org/order/33/"
    }
}







Payments


Recurring


Tarieven


Developers


Partners



Features

Overzicht


Dashboard


Checkout


Security



Support

Help Center


Betaald met Mollie


Status


Changelog


Contact support



Bedrijf

Over Mollie

**Comment by ywarnier** [comment]
That's great. So as I said: either you develop the extension to the buycourses plugin yourself, with that information, or yu ask a team of developers to do it. This is how it was done for bank payment, PayPal and Culqi. The Chamilo Association has no direct interest in having people sell courses, as this goes against its goal of "Making quality education more accessible to all".

**Comment by learningcentrum** [comment]
You are talking about Culqi. in don't have it on my buycourses, I haven only bank and paypal
I will try to do it

**Comment by ywarnier** [comment]
To quote myself above:
"The easiest process is probably to look at the difference (in the 1.11.x branch) between PayPal and Culqi in the plugin/buycourses/ directory"
and
"1.11.x is the development version that will result (soon) in the publication of 1.11.4."

Please refer to the README file in the 1.11.x branch for more info, and read the wiki documentation if you want to contribute.

**Comment by ywarnier** [comment]
This remains on hold until a lot of community members are asking for it *or* a sponsor can finance this feature.

**Comment by mitdrissia** [comment]
I am also interested in the ideal payment. Is it allready included in the new version

**Comment by learningcentrum** [comment]
we working on it, can you give some help ?

**Comment by mitdrissia** [comment]
I am not sure how i can help. If we contribute and hire a freelancer to do
it for us that would be great.

**Comment by ywarnier** [comment]
You can also contact official providers to do it for you and integrate it into the next version of Chamilo: https://chamilo.org/en/providers/
That has the special advantage of ensuring that gets into Chamilo (instead of being done for you and outdated at the next update). If you don't, please ensure the freelancer respects Chamilo's coding conventions (see wiki above) and that (s)he contributes the code back to Chamilo so we can check it for security flaws.

**Comment by learningcentrum** [comment]
Hallo
we our now finding out how its work, not simple
we most make the connector to the plug in buy courses

greetings
Martin Doornbos

 
Op 22-09-2018 om 19:17 uur schreef Yannick Warnier <notifications@github.com>:
You can also contact official providers to do it for you and integrate it into
the next version of Chamilo: https://chamilo.org/en/providers/
That has the special advantage of ensuring that gets into Chamilo (instead of
being done for you and outdated at the next update). If you don't, please
ensure the freelancer respects Chamilo's coding conventions (see wiki above)
and that (s)he contributes the code back to Chamilo so we can check it for
security flaws.
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.


---
Dit e-mailbericht is gecontroleerd op virussen met Avast antivirussoftware.
https://www.avast.com/antivirus

**Comment by mitdrissia** [comment]
Yes, but i will contact mollie. Also how about stripe integration? It
automatically has ideal,credit card and many other payment gateways
integrated. Have you contacted them before?

**Comment by learningcentrum** [comment]
I have made a shop in the mother website www.learningcentrum.nl

**Comment by ywarnier** [comment]
Not sure if this is relevant anymore. Been 6 years and Stripe is availble. Closing.
